Draw.it.Server/Hubs/GameplayHub.cs

```diff
@@ -93,12 +97,28 @@ public async Task SendMessage(string message)
     public async Task SendDraw(DrawDto drawDto)
     {
         var user = await ResolveUserAsync();
+        var roomId = user.RoomId!;
+        var game = _gameService.GetGame(roomId);
+
+        if (user.Id != game.CurrentDrawerId)
+        {
+            return;
+        }
+
         await Clients.GroupExcept(user.RoomId!, Context.ConnectionId).SendAsync("ReceiveDraw", drawDto);
     }
 
     public async Task SendClear()
     {
         var user = await ResolveUserAsync();
+        var roomId = user.RoomId!;
+        var game = _gameService.GetGame(roomId);
+
+        if (user.Id != game.CurrentDrawerId)
+        {
+            return;
+        }
+
         await Clients.GroupExcept(user.RoomId!, Context.ConnectionId).SendAsync("ReceiveClear");
```

Comment:
You should use an already existing variable here

```suggestion
        await Clients.GroupExcept(roomId, Context.ConnectionId).SendAsync("ReceiveClear");
```